tests/unit/bokeh/core/property/test_validation__property.py

```diff
@@ -175,7 +175,27 @@ def test_Dict(self) -> None:
         p = Dict(String, Float)
         with pytest.raises(ValueError) as e:
             p.validate("junk")
-        assert matches(str(e.value), r"expected an element of Dict\(String, Float\), got 'junk'")
+        assert matches(str(e.value), r"expected a dict of type Dict\(String, Float\), got a value of type <class 'str'>")
+
+    def test_Dict_Invalid_Key(self) -> None:
+        d = Dict(String, String)
+        with pytest.raises(ValueError) as err:
+            d.validate({"Foo": "Bar", 1: "Baz"})
+        assert "invalid keys: 1" in str(err.value)
+
+    def test_Dict_Invalid_Value(self) -> None:
+        d = Dict(String, String)
+        with pytest.raises(ValueError) as err:
+            d.validate({"Foo": "Bar", "Baz": 1})
+        assert "values for keys: Baz" in str(err.value)
+
+    def test_Dict_Valid(self) -> None:
+        d = Dict(String, String)
+        try:
+            d.validate({"Foo": "Bar"})
```

Comment:
We could also consider really expanding the validation of correct cases via pytests `parametrize` decorator, something like (untested):
```suggestion
    KEYS = ((String, "Foo"), ("Int", 10))
    VALS = ((String, "Bar"), ("Int", 20), (Float, 1.23), (List[Float], [1.23, 3.45]))

    @pytest.mark.parametrize("key_type,key", KEYS)
    @pytest.mark.parametrize("val_type,val", VALS)
    def test_Dict_Valid(self, key_type, key, val_type, val) -> None:
        d = Dict(key_type, val_type)
        try:
            d.validate({key: val})
```

Reply:
Sounds interesting, I'll give it a shot